jetty-util/src/test/java/org/eclipse/jetty/util/thread/ReservedThreadExecutorTest.java

```diff
@@ -332,6 +359,6 @@ public void run()
         pool.stop();
 
         assertThat(usedReserved.get() + usedPool.get(), is(LOOPS));
```

Comment:
My original stress test repeated the above processing a few times because the `usedReserved` counter would stick to 0 after the bug triggered, so I also tested that `usedReserved` must be > 0.

Reply:
This does not check that `tryExecute` doesn't stop working after some time. You need to reset the counters and latch after those assertions and loop over to resubmit the tasks.